Sources/TSCBasic/Process.swift

```diff
@@ -677,7 +680,10 @@ public final class Process {
         var stdinPipe: [Int32] = [-1, -1]
         try open(pipe: &stdinPipe)
 
-        let stdinStream = try LocalFileOutputByteStream(filePointer: fdopen(stdinPipe[1], "wb"), closeOnDeinit: true)
+        guard let fp = fdopen(stdinPipe[1], "wb") else {
+            throw Process.Error.stdinUnavailable
+        }
+        let stdinStream = try LocalFileOutputByteStream(filePointer: fp, closeOnDeinit: true)
```

Comment:
I'd prefer to not change this on 5.10, ie. force unwrap like in the other cases.

Reply:
The whole point of adding a `guard let` here is that it's much safer and actually checks for the optional that the recent version of Bionic passes in. Why go back to a force unwrap then?

Reply:
Yeah, I kind of agree with keeping the guard-let here too. If it's about accepting risk, then a force-unwrap would be a worse outcome than an error message.

Reply:
The bar for 5.10 is extremely high at this point. If you'd like this change in, it needs to have zero impact outside of fixing the compilation errors. I'm all for the guard on main (and ideally, these would all be actually checks instead of force unwraps).

Reply:
I think this passes that bar: the guard-let's are obviously fine, as they only make the code safer. The two remaining force unwraps: one is of a `baseAddress` access, which should never fail unless someone passes in a nil to that internal function intentionally, and the other is in a test support function.

This patch was merged into the main branch a month ago and we've had zero problems with it: I think the same will be true on 5.10 and it will keep this toolchain compiling on linux with the newest versions of glibc.

Reply:
Sorry, I mixed up which thing this branch was targeting. Short of finding something like a security bug or memory issue like that, the changes can't have any impact on macOS at all. Even changing when something throws or aborts is more than what is acceptable for the 5.10 branch at this point. Sorry about that.

Reply:
I think this passes that bar too, as these were all raw accesses on macOS before, so if they were going to fail, _they would fail regardless_. All we're changing on macOS is the way that fails, and I don't think that qualifies as "Even changing when something throws or aborts" as technically these all fail at _exactly the same point_ on macOS after this patch, they just give better errors and fix compilation with glibc and bionic.